### PR TITLE
Add combinators fromEitherM and fromOptionM #668

### PR DESCRIPTION
**Pull request by Vilkina**

### CODE REVIEW

File: interop-java/jvm/src/test/scala/scalaz/zio/interop/javaconcurrent.scala [hunk 1]
```diff
@@ -25,27 +25,29 @@ import scala.concurrent.ExecutionException
 
 object javaconcurrent {
 
-  implicit class IOObjJavaconcurrentOps(private val taskObj: Task.type) extends AnyVal {
+  implicit class IOObjJavaconcurrentOps(private val ioObj: IO.type) extends AnyVal {
```
Comment (iravid) [inline interop-java/jvm/src/test/scala/scalaz/zio/interop/javaconcurrent.scala]: How about we move these changes to a separate pull request?

Comment (iravid) [reply interop-java/jvm/src/test/scala/scalaz/zio/interop/javaconcurrent.scala]: Either way - let's keep this on the `Task` object, as all Java futures may fail with a throwable.

Comment (Vilkina) [reply interop-java/jvm/src/test/scala/scalaz/zio/interop/javaconcurrent.scala]: Again :(
